### DISCUSSION TITLE
Time to full or empty

### DISCUSSION THREAD

**Discussion by vdvmichel** [TypQxQ/Sigenergy-Home-Assistant-Integration#32] [Ideas]
I it would be nice to see what the remaining time would be until the battery is full or empty...

**Comment by vdvmichel** [comment]
Example I found…
https://community.home-assistant.io/t/solar-battery-run-time-till-empty/408778/67

**Comment by micvppl** [comment]
ChatGPT says:

To calculate the time it will take for a battery to get empty based on power usage, you can create a **template sensor** in Home Assistant. Below is an example configuration you can add to your `configuration.yaml` file.

### Steps:

1. **Determine the required inputs:**
   - `battery_capacity` (e.g., in watt-hours (Wh)).
   - `current_power_usage` (e.g., in watts (W)).

2. **Create a sensor to calculate time remaining.**

Here’s the code:

```yaml
template:
  - sensor:
      - name: "Battery Time Remaining"
        unit_of_measurement: "hours"
        state: >
          {% set battery_capacity = states('sensor.battery_capacity') | float(0) %}
          {% set power_usage = states('sensor.current_power_usage') | float(0) %}
          {% if power_usage > 0 %}
            {{ (battery_capacity / power_usage) | round(2) }}
          {% else %}
            0
          {% endif %}
        availability: >
          {{ states('sensor.battery_capacity') | is_number and
             states('sensor.current_power_usage') | is_number }}
```

### Explanation:
1. **Inputs:**
   - `sensor.battery_capacity`: Represents the current battery capacity in Wh.
   - `sensor.current_power_usage`: Represents the current power draw in W.

2. **Calculation Logic:**
   - Time to empty = Battery Capacity (Wh) ÷ Power Usage (W).
   - It handles cases where power usage might be zero to avoid division errors.

3. **Availability:**
   - Ensures the calculation only works when the input sensors provide numeric values.

### Example Data Sources:
- `sensor.battery_capacity`: Create this sensor to track your battery's current capacity (e.g., via integration with your inverter or battery management system).
- `sensor.current_power_usage`: Often provided by an energy monitoring device like a smart plug, inverter, or energy meter.

### Customization:
- Replace `sensor.battery_capacity` and `sensor.current_power_usage` with the actual names of your sensors.
- Adjust units if your sensors provide values in other units (e.g., kWh or kW).

This template sensor will dynamically update and show the time remaining until the battery is depleted based on real-time power usage.

**Comment by TypQxQ** [comment]
It's a verry nice idea!
And I love you used ChatGPT!
Will implement for sure!

Regards, Andrei
